yt_dlp/extractor/ceskatelevize.py

```diff
@@ -106,67 +127,94 @@ def _real_extract(self, url):
         if playlist_description:
             playlist_description = playlist_description.replace('\xa0', ' ')
 
-        type_ = 'IDEC'
+        type_ = 'episode'
+        is_live = False
         if re.search(r'(^/porady|/zive)/', parsed_url.path):
             next_data = self._search_nextjs_data(webpage, playlist_id)
             if '/zive/' in parsed_url.path:
                 idec = traverse_obj(next_data, ('props', 'pageProps', 'data', 'liveBroadcast', 'current', 'idec'), get_all=False)
+                sidp = traverse_obj(next_data, ('props', 'pageProps', 'data', 'liveBroadcast', 'current', 'showId'), get_all=False)
+                is_live = True
             else:
                 idec = traverse_obj(next_data, ('props', 'pageProps', 'data', ('show', 'mediaMeta'), 'idec'), get_all=False)
                 if not idec:
                     idec = traverse_obj(next_data, ('props', 'pageProps', 'data', 'videobonusDetail', 'bonusId'), get_all=False)
                     if idec:
                         type_ = 'bonus'
+                sidp = self._search_regex(r'https?://(?:www\.)?ceskatelevize\.cz/(?:ivysilani|porady|zive)/([0-9]+)-', url, playlist_id, default=playlist_id)
             if not idec:
                 raise ExtractorError('Failed to find IDEC id')
-            iframe_hash = self._download_webpage(
-                'https://www.ceskatelevize.cz/v-api/iframe-hash/',
-                playlist_id, note='Getting IFRAME hash')
-            query = {'hash': iframe_hash, 'origin': 'iVysilani', 'autoStart': 'true', type_: idec, }
+            sidp = sidp.rsplit('-')[0]
+            query = {'origin': 'iVysilani', 'autoStart': 'true', 'sidp': sidp, type_: idec}
```

Comment:
```suggestion
```
you could inline these in `_download_webpage`
```py
            webpage = self._download_webpage(
                'https://player.ceskatelevize.cz/', playlist_id,
                note='Downloading player', query={
                    'origin': 'iVysilani',
                    'autoStart': 'true',
                    'sidp': sidp.rsplit('-')[0],
                    type_: idec,
                })
```

Reply:
OK

Reply:
The page that I got for this contained a link to `/config/playerConfig.js` where these are defined:

```json
{
...
  "PLAYLIST_LEGACY_URI":"https://www.ceskatelevize.cz/ivysilani/ajax/get-client-playlist",
  "PLAYLIST_LIVE_URI":"https://api.ceskatelevize.cz/video/v1/playlist-live/v1",
  "PLAYLIST_VOD_URI":"https://api.ceskatelevize.cz/video/v1/playlist-vod/v1",
...
}
```
Should these be used with the extractor's hard-coded values as default?

Also, should the `VOD` API be used rather than the `LEGACY` API?